ports/glutin/window.rs

```diff
@@ -344,61 +357,114 @@ impl Window {
         GlRequest::Specific(Api::OpenGlEs, (3, 0))
     }
 
-    fn handle_window_event(&self, event: glutin::Event) -> bool {
-        match event {
-            Event::ReceivedCharacter(ch) => {
-                if !ch.is_control() {
-                    self.pending_key_event_char.set(Some(ch));
-                }
+    #[cfg(not(target_os = "windows"))]
+    fn handle_received_character(&self, ch: char) {
+        if !ch.is_control() {
+            self.pending_key_event_char.set(Some(ch));
+        }
+    }
+
+    #[cfg(target_os = "windows")]
+    fn handle_received_character(&self, ch: char) {
+        let modifiers = Window::glutin_mods_to_script_mods(self.key_modifiers.get());
+        if let Some(last_pressed_key) = self.last_pressed_key.get() {
+            let event = WindowEvent::KeyEvent(Some(ch), last_pressed_key, KeyState::Pressed, modifiers);
+            self.event_queue.borrow_mut().push(event);
+        } else {
+            // Only send the character if we can print it (by ignoring characters like backspace)
+            if ch >= ' ' {
+                let event = WindowEvent::KeyEvent(Some(ch),
+                                                  Key::A /* unused */,
```

Comment:
Duplicate https://dxr.mozilla.org/servo/rev/065f50014f321466c979120967da720e88ae2b29/components/script/dom/keyboardevent.rs#344-438 and ignore the stuff about key location.